classes/profile.php

```diff
@@ -80,6 +83,21 @@ protected function get_flamedatad3(): string {
         return json_decode($this->get_flamedatad3json());
     }
 
+    /**
+     * Set the parameters field. Will be truncated if longer than MAX_STR_LEN bytes.
+     *
+     * @param string $parameters
+     * @throws \coding_exception
+     */
+    protected function set_parameters(string $parameters) {
+        /* The database field can only store MAX_STR_LEN bytes. So if the string is longer, we chop it off and add
+         * an ellipsis. */
+        if (strlen($parameters) > self::MAX_STR_LEN) {
+            $parameters = substr($parameters, 0, self::MAX_STR_LEN - 4) . " ...";
```

Comment:
Hmm, I think a better approach here would be to use the `…` single character instead of using 4 characters. 
- There's no need to have a space between the ellipsis and the string
- You can display up to 3 more characters :)

So you will need to adjust the numbers and characters around, otherwise it seems like the approach is fine.

Reply:
Changed DB field to TEXT